ludwig/backend/deepspeed.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict, Optional, Type
+
+import deepspeed
+
+from ludwig.backend.base import DataParallelBackend
+from ludwig.constants import FALLBACK_BATCH_SIZE
+from ludwig.distributed import init_dist_strategy
+from ludwig.utils.batch_size_tuner import BatchSizeEvaluator
+
+
+class DeepSpeedBackend(DataParallelBackend):
```

Comment:
How does this new backend interact with Ray/Ray Datasets? Noticed that DataParallelBackend uses the LocalPreprocessingMixin, so curious what you're thinking here.

Reply:
It works the same as the existing HorovodBackend. You run with the command line tool `deepspeed`. It processes the data in-memory (like local training). It doesn't use Ray (that's the DeepSpeedStrategy).

Reply:
Nice, just read the PR description. Thanks!